app/controllers/orders_controller.rb

```diff
@@ -1,25 +1,34 @@
 class OrdersController <ApplicationController
+  def index
+    user = User.find(session[:user])
+    @orders = user.orders
+  end
 
   def new
-
+    if !current_user
```

Comment:
This is a sad path for visitors who click checkout without being logged in or registered